File: server/game-service/src/main/java/com/chess/gameservice/game/turn/GameTurn.java
package com.chess.gameservice.game.turn;

import com.chess.gameservice.game.piece.PieceType;
import com.chess.gameservice.game.player.PlayerColor;
import com.chess.gameservice.game.position.Position;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@Builder
@AllArgsConstructor
public class GameTurn {
    Position initialPosition;
    Position destinationPosition;
    PieceType pieceType;
    PlayerColor playerColor;
}

File: app/src/interfaces/History/GameTurn.ts
import { BoardPosition } from '../Game/BoardPosition';
import { PieceType } from '../Game/Piece';
import { PlayerColor } from '../Game/Player';

export interface GameTurn {
  turnId: number;
  initialPosition: BoardPosition;
  destinationPosition: BoardPosition;
  pieceType: PieceType;
  playerColor: PlayerColor;
  turnNumber: number;
}
